pandas/core/series.py

```diff
@@ -5629,6 +5636,124 @@ def between(
 
         return lmask & rmask
 
+    def case_when(
+        self,
+        caselist: list[tuple[ArrayLike, ArrayLike | Scalar]],
+    ) -> Series:
+        """
+        Replace values where the conditions are True.
+
+        Parameters
+        ----------
+        caselist : A list of tuples of conditions and expected replacements.
+            Takes the form:  ``(condition0, replacement0)``,
+            ``(condition1, replacement1)``, ... .
+            ``condition`` should be a 1-D boolean array-like object
+            or a callable. If ``condition`` is a callable,
+            it is computed on the Series
+            and should return a boolean Series or array.
```

Comment:
I think a callable would work, but then  you would need two changes:
1. Change the type of `caselist` to allow the first item in the tuple to be `ArrayLike | Callable[[Series], Series | np.ndarray | Sequence[bool]]`  (I think that is right)
2. One of the tests should be testing a callable as well.

E.g., I think this would work based on my read of the code.
```python
df = pd.DataFrame({
        ...     "a": [0,0,1,2],
        ...     "b": [0,3,4,5],
        ...     "c": [6,7,8,9]
        ... })
df.c.case_when(caselist=[(lambda s: s > 7, df.a)])
```
NOTE:  `DataFrame.mask()` and `Series.mask()` accept a callable, but the typing doesn't exist in the pandas source, and `pandas-stubs` doesn't reflect that either.